### PR TITLE
[fix](decimal) fix cast rounding half up with negative number

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

before
```
mysql [(none)]>select cast(-280.00000000 as decimal(16,4));
+-----------------------------------------+
| cast(-280.00000000 as DECIMALV3(16, 4)) |
+-----------------------------------------+
|                               -279.9999 |
+-----------------------------------------+
```

after

```
mysql [(none)]>select cast(-280.00000000 as decimal(16,4));
+-----------------------------------------+
| cast(-280.00000000 as DECIMALV3(16, 4)) |
+-----------------------------------------+
|                               -280.0000 |
+-----------------------------------------+

mysql [(none)]>select cast(-280.00005000 as decimal(16,4));
+-----------------------------------------+
| cast(-280.00005000 as DECIMALV3(16, 4)) |
+-----------------------------------------+
|                               -280.0001 |
+-----------------------------------------+
```

mysql

image

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.91 seconds
 stream load tsv:          511 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17161958127 Bytes

File: be/src/vec/data_types/data_type_decimal.h [hunk 1]
```diff
@@ -505,7 +505,11 @@ void convert_decimal_cols(
         MaxNativeType multiplier =
                 DataTypeDecimal<MaxFieldType>::get_scale_multiplier(scale_from - scale_to);
         for (size_t i = 0; i < sz; i++) {
-            vec_to[i] = (vec_from[i] + multiplier / 2) / multiplier;
+            if (vec_from[i] >= 0) {
```
Comment (Gabriel39) [inline be/src/vec/data_types/data_type_decimal.h:508]:
```suggestion
            vec_to[i] = (vec_from[i] + multiplier / 2 * (vec_from[i] >= 0)) / multiplier;
```

**Review by Gabriel39 — CHANGES_REQUESTED** [review]

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**